contracts/src/chainlink/solidity/emergency/StarknetValidator.sol

```diff
@@ -176,8 +181,13 @@ contract StarknetValidator is TypeAndVersionInterface, AggregatorValidatorInterf
 
   /// @notice L1 oracle is asked for a fast L1 gas price, and the price multiplied by the configured gas estimate
   function _approximateFee() internal view returns (uint256) {
+    uint256 gasPrice = approximateGasPrice();
+    return gasPrice * s_gasConfig.gasEstimate;
+  }
+
+  function approximateGasPrice() public view returns (uint256) {
```

Comment:
We now changed the API surface of our contract, just to test...

A better way to do this would be to keep the method internal, then extend the contract with another (test contract) and there expose the functionality we need to test.

We're short on time, so leave it as is, but please at least add a comment to this public fn.

Reply:
Might be useful for monitoring, so let's leave it in.